dvc/api.py

```diff
@@ -113,10 +109,7 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
 
 @contextmanager
 def _make_repo(repo_url, rev=None):
-    if not repo_url or os.path.exists(repo_url):
-        assert (
-            rev is None
-        ), "Git revisions are not supported for local DVC projects."
+    if rev is None and (not repo_url or os.path.exists(repo_url)):
         yield Repo(repo_url)
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
```

Comment:
Why is `external_repo()` used if there's a `rev`? What if I want to access a file in the current DVC repo but from a previous commit? I'm trying to test this but I'm getting `AttributeError: module 'dvc' has no attribute 'api'` when running:

```py
import dvc

print(dvc.api.read('data'))
```
Using DVC 0.87.0 installed with `pip` in a py3 virtualenv.

@Suor @skshetry @shcheklein

Reply:
UPDATE: Nevermind about the Python error. I just had to `import dvc.api`

Reply:
@jorgeorpinel, it should just clone the current repo and checkout to given `rev` and access that file.

Reply:
Testing this, I see it works locally anyway. But is it using `external_repo()` to clone the current repo in a temporary folder and then use that instead of just the current Git repo's tree?

Reply:
p.s. my test script:

```py
import dvc.api

print(dvc.api.read('data', rev='HEAD^'))
```
`data` should be a DVC-tracked file with 2 different consecutive versions (in `HEAD^` and in `HEAD`).

Reply:
My point is I think `Repo()` also accepts `rev`, just wondering why it's not passed, and whether that would be a better implementation.

Reply:
`Repo` does not accept `rev`. Repo is all about current workspace, so, you cannot just checkout the user's workspace and modify/read. So, even though it's still is a `Repo` underneath (if it's a DVC repo), it's cloned, checked out to a `rev`, put into a `tmp` cache directory and operated inside of it.

Reply:
Got it. I was just confused by the name `external_repo` then. I guess that's the only way to get a previous rev from the current repo unfortunately.